File: lib/result/EdScan.java
package com.teamlab.communication.wireless.lpr9201.result;

import java.util.HashMap;
import java.util.Map;

/**
 * リザルト EdScan
 */
public class EdScan extends Result {
	/**
	 * 
	 */
	public static final int RESULT_CODE = 0x8C;

	/**
	 * 
	 */
	public static final int DATA_LENGTH_SIZE = 1; // byte

	private static final int RATE1_LENGTH = 0x1D;
	private static final int RATE2_LENGTH = 0x0E;
	private static final int START_CH = 33;

	private Map<Integer, Integer> value = new HashMap<Integer, Integer>();

	public EdScan(int[] datas) {
		super(RESULT_CODE, DATA_LENGTH_SIZE, datas);

		if (datas.length == RATE1_LENGTH) {
			for (int i = 0; i < datas.length; i++) {
				value.put(START_CH + i, datas[i]);
			}
		} else if (datas.length == RATE2_LENGTH) {
			int offset = 0;
			for (int i = 0; i < datas.length; i++) {
				value.put(START_CH + offset + i, datas[i]);
				value.put(START_CH + offset + i + 1, datas[i]);
				offset++;
			}
		}
	}

	public Map<Integer, Integer> getRssis() {
		return this.value;
	}
}

File: lib/result/ed_scan.js
/* jslint node: true */
"use strict"

/**
 * リザルト EdScan
 */

exports.EdScan = EdScan;

/**
 * 結果コード
 */
EdScan.RESULT_CODE = 0x8C;

/**
 * データ長バイトサイズ
 */
EdScan.DATA_LENGTH_BYTE_SIZE = 1; // byte


/**
 * Rate1のチャンネル数
 */
var RATE1_LENGTH = 0x1D;

/**
 * Rate2のチャンネル数
 */
var RATE2_LENGTH = 0x0E;

/**
 * 開始チャンネル番号
 */
var START_CH = 33;


/**
 * @param {Result} result
 */
function EdScan(result) {
  this.result = result;
  this.value = {};

  if (result.datas.length === RATE1_LENGTH) {
    for (var i = 0; i < result.datas.length; i++) {
      this.value[START_CH + i] = result.datas[i];
    }
  } else if (result.datas.length === RATE2_LENGTH) {
    var offset = 0;
    for (var i = 0; i < result.datas.length; i++) {
      this.value[START_CH + offset + i] = result.datas[i];
      this.value[START_CH + offset + i + 1] = result.datas[i];
      offset++;
    }
  }
}

/**
 * パース可能か
 * @param {Result} result
 */
EdScan.canParse = function(result) {
  return result !== null && result.resultCode === EdScan.RESULT_CODE;
}
